src/Utils/logger.c

```diff
@@ -0,0 +1,104 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdarg.h>
+#include <time.h>
+#include <pthread.h>
+#include <syslog.h>
+#include "logger.h"
+
+#define LOG_FILE_PATH "/var/log/pproc.log"
+
+static FILE *log_file = NULL;
+static LogLevel console_log_level = LL_INFO;
+static LogLevel file_log_level = LL_INFO;
+static pthread_mutex_t log_mutex = PTHREAD_MUTEX_INITIALIZER;
+
+static const char *level_strings[] = {
+    "ERROR",
+    "WARNING",
+    "INFO",
+    "DEBUG"};
+
+void init_logger(const char *log_file_path, LogLevel console_level, LogLevel file_level)
+{
+    console_log_level = console_level;
+    file_log_level = file_level;
+
+    if (log_file_path)
+    {
+        log_file = fopen(log_file_path, "a");
+        if (!log_file)
+        {
+            fprintf(stderr, "Failed to open log file: %s\n", log_file_path);
+        }
+    }
+
+    openlog("pproc", LOG_PID | LOG_CONS, LOG_DAEMON);
+}
+
+void cleanup_logger(void)
+{
+    if (log_file)
+    {
+        fclose(log_file);
+        log_file = NULL;
+    }
+    closelog();
+}
+
+void log_message(LogLevel level, const char *format, ...)
+{
+    pthread_mutex_lock(&log_mutex);
+
+    time_t now;
+    time(&now);
+    char timestamp[26];
+    ctime_r(&now, timestamp);
+    timestamp[24] = '\0'; // Remove newline
+
+    va_list args;
```

Comment:
may want to move all declarations to top to match more common C standard but this is very minor